server/src/main/kotlin/com/kamelia/hedera/rest/file/Service.kt

```diff
@@ -18,25 +19,32 @@ import com.kamelia.hedera.rest.token.PersonalToken
 import com.kamelia.hedera.rest.user.User
 import com.kamelia.hedera.rest.user.UserRole
 import com.kamelia.hedera.util.FileUtils
+import com.kamelia.hedera.util.toUUID
+import com.kamelia.hedera.util.toUUIDShort
 import com.kamelia.hedera.util.uuid
 import io.ktor.http.*
 import io.ktor.http.content.*
 import java.util.*
 import kotlin.math.ceil
 
 val CUSTOM_LINK_REGEX = """^[a-z0-9]+(-[a-z0-9]+)*$""".toRegex()
+val PERSONAL_TOKEN_REGEX = """^[a-f0-9]{64}$""".toRegex()
 
 object FileService {
 
     suspend fun handleFileWithToken(
         part: PartData.FileItem,
         creatorToken: String
     ): ActionResponse<FileRepresentationDTO> = Connection.transaction {
-        val token = PersonalToken.findByToken(creatorToken) ?: throw ExpiredOrInvalidTokenException()
+        if (!PERSONAL_TOKEN_REGEX.matches(creatorToken)) throw ExpiredOrInvalidTokenException()
 
-        if (token.deleted) throw ExpiredOrInvalidTokenException()
+        val (tokenId, token) = creatorToken.substring(0, 32) to creatorToken.substring(32)
```

Comment:
do that in two lines, no need to create a pair just to destructure it right after. it hurts the readability a bit too tbh